src/test/java/com/google/devtools/build/lib/actions/ActionCacheCheckerTest.java

```diff
@@ -999,6 +1032,44 @@ public void saveOutputMetadata_treeMetadata_localArchivedArtifactLoaded() throws
     assertThat(metadataHandler.getTreeArtifactValue(output)).isEqualTo(expectedMetadata);
   }
 
+  @Test
+  public void saveOutputMetadata_treeFileExpired_treeMetadataNotLoaded() throws Exception {
```

Comment:
Also add a test for when the archived representation is expired?

Reply:
Done.